### ISSUE TITLE
Weighted Gradient Descent

### ISSUE THREAD

**Issue by francescodeaglio** [eth-easl/modyn#217]
At the moment, only plain minibatch gradient descent is supported. Typically, Coreset methods assign weights to each sample and this weight is used in backprop.
